libs/platform/src/lib/components/form/multi-combobox/commons/base-multi-combobox.ts

```diff
@@ -0,0 +1,793 @@
+        if (!isOpen) {
+            this.searchTermChanged('');
+        }
+
+        this.showList(!isOpen);
+
+        if (this.isOpen && this.listComponent) {
+            this.listComponent.setItemActive(0);
+        }
+    }
+
+    /**
+     * Handle Keydown on Input
+     * @hidden
+     */
+    onInputKeydownHandler(event: KeyboardEvent): void {
+        if (this.readonly) {
+            return;
+        }
+
+        if (KeyUtil.isKeyCode(event, DOWN_ARROW)) {
+            event.preventDefault();
+
+            if (event.altKey) {
+                this.showList(true);
+            }
+
+            if (this.isOpen && this.listComponent) {
+                this.listComponent.setItemActive(0);
+            } else if (!this.isOpen) {
+                this._chooseOtherItem(1);
+            }
+        } else if (KeyUtil.isKeyCode(event, UP_ARROW)) {
+            event.preventDefault();
+
+            this._chooseOtherItem(-1);
+        } else if (KeyUtil.isKeyCode(event, ENTER)) {
+            this.toggleSelectionByInputText();
+        } else if (KeyUtil.isKeyCode(event, ESCAPE)) {
+            event.stopPropagation();
+
+            this.showList(false);
+        } else if (!event.ctrlKey && !KeyUtil.isKeyCode(event, this._nonOpeningKeys)) {
+            this.showList(true);
+            const acceptedKeys =
+                !KeyUtil.isKeyCode(event, BACKSPACE) &&
+                !KeyUtil.isKeyType(event, 'alphabetical') &&
+                !KeyUtil.isKeyType(event, 'numeric');
+            if (this.isEmptyValue && acceptedKeys) {
+                this.listComponent?.setItemActive(0);
+            }
+        }
+    }
+
+    /** Method passed to list component */
+    handleListFocusEscape(direction: FocusEscapeDirection): void {
+        if (direction === 'up') {
+            this.searchInputElement.elementRef().nativeElement.focus();
+        }
+    }
+
+    /** @hidden */
+    protected get ds(): ListDataSource<any> {
+        return <ListDataSource<any>>this.dataSource;
+    }
+
+    /** @hidden
+     * Method that picks other value moved from current one by offset, called only when Multi Combobox is closed */
+    private _chooseOtherItem(offset: number): void {
+        if (this._selected?.length === this._flatSuggestions.length) {
+            this.inputText = '';
+            return;
+        }
+
+        const activeValue: SelectableOptionItem = this._getSelectItemByInputValue(this.inputText);
+        const index: number = this._flatSuggestions.findIndex(value => value === activeValue);
+        let item: SelectableOptionItem;
+        if (!this.inputText && offset === -1) {
+            item = this._flatSuggestions[this._flatSuggestions.length - 1];
+        } else {
+            item = this._flatSuggestions[index + offset];
+        }
+
+        if (item) {
+            this.setInputTextFromOptionItem(item);
+        }
+
+        const selectedIndex = this._selected.findIndex(value => value.label === item.label);
```

Comment:
what happens if `item` doesn't exist?

Reply:
@Betrozov nothing _(`selectedIndex` will be `-1` and the function do not choose any item)_

Reply:
@Betrozov with `item?.label` nothing _(`selectedIndex` will be `-1` and the function do not choose any item)_
Thanks, fixed `item?.label`